src/hotspot/share/memory/dynamicArchive.cpp

```diff
@@ -367,6 +368,15 @@ void DynamicArchive::dump() {
     return;
   }
 
+  // regenerate lambdaform holder classes
+  if (LambdaFormInvokers::should_regenerate_holder_classes()) {
+    JavaThread* THREAD = JavaThread::current();
+    assert(THREAD->is_Java_thread(), "Should be JavaThread");
```

Comment:
JavaThread::current() already asserts this.

Reply:
Will remove the assert.
I will also move the check should_regenerate_holder_classes to LambdaFormInvokers::should_regenerate_holder_classes().